pkg/cmd/attestation/inspect/inspect.go

```diff
@@ -93,96 +88,260 @@ func NewInspectCmd(f *cmdutil.Factory, runF func(*Options) error) *cobra.Command
 				apiClient := api.NewLiveClient(hc, opts.Hostname, opts.Logger)
 				td, err := apiClient.GetTrustDomain()
 				if err != nil {
-					return err
+					return fmt.Errorf("error getting trust domain, make sure you are authenticated against the host: %w", err)
 				}
-				tenant, found := ghinstance.TenantName(opts.Hostname)
+				_, found := ghinstance.TenantName(opts.Hostname)
 				if !found {
-					return fmt.Errorf("Invalid hostname provided: '%s'",
+					return fmt.Errorf("invalid hostname provided: '%s'",
 						opts.Hostname)
 				}
 
 				config.TrustDomain = td
-				opts.Tenant = tenant
 			}
 
 			opts.SigstoreVerifier = verification.NewLiveSigstoreVerifier(config)
 
+			if runF != nil {
+				return runF(opts)
+			}
+
 			if err := runInspect(opts); err != nil {
 				return fmt.Errorf("Failed to inspect the artifact and bundle: %w", err)
 			}
 			return nil
 		},
 	}
 
-	inspectCmd.Flags().StringVarP(&opts.BundlePath, "bundle", "b", "", "Path to bundle on disk, either a single bundle in a JSON file or a JSON lines file with multiple bundles")
-	inspectCmd.MarkFlagRequired("bundle") //nolint:errcheck
 	inspectCmd.Flags().StringVarP(&opts.Hostname, "hostname", "", "", "Configure host to use")
-	cmdutil.StringEnumFlag(inspectCmd, &opts.DigestAlgorithm, "digest-alg", "d", "sha256", []string{"sha256", "sha512"}, "The algorithm used to compute a digest of the artifact")
 	cmdutil.AddFormatFlags(inspectCmd, &opts.exporter)
 
 	return inspectCmd
 }
 
-func runInspect(opts *Options) error {
-	artifact, err := artifact.NewDigestedArtifact(opts.OCIClient, opts.ArtifactPath, opts.DigestAlgorithm)
-	if err != nil {
-		return fmt.Errorf("failed to digest artifact: %s", err)
-	}
+type BundleInspectResult struct {
+	InspectedBundles []BundleInspection `json:"inspectedBundles"`
+}
+
+type BundleInspection struct {
+	Authentic              bool                  `json:"authentic"`
+	Certificate            CertificateInspection `json:"certificate"`
+	TransparencyLogEntries []TlogEntryInspection `json:"transparencyLogEntries"`
+	SignedTimestamps       []time.Time           `json:"signedTimestamps"`
+	Statement              *in_toto.Statement    `json:"statement"`
+}
+
+type CertificateInspection struct {
+	certificate.Summary
+	NotBefore time.Time `json:"notBefore"`
+	NotAfter  time.Time `json:"notAfter"`
+}
 
-	opts.Logger.Printf("Verifying attestations for the artifact found at %s\n\n", artifact.URL)
+type TlogEntryInspection struct {
+	IntegratedTime time.Time
+	LogID          string
+}
 
+func runInspect(opts *Options) error {
+	// 1. Load the bundles
+	// 2. Parse & check the "authenticity" of the bundle
+	// 3. Summarize the certs
```

Comment:
**Is there any concern about these being crufty relative to the logic that follows?**

I just assume to separate these out as one-line comments before each of the relevant code sections below.